file_stream/file_stream.go

```diff
@@ -0,0 +1,111 @@
+package file_stream
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type FileStream struct {
+	file_name string
+	reader    *os.File
+	watcher   *fsnotify.Watcher
+
+	// Used for get command completion notifications
+	Done chan bool
+
+	// Used to abort streaming when a client disconnects, etc
+	ctx context.Context
+}
+
+//-------------------------------------------------------------------------------------------------
+func NewFileStream(file_name string, ctx context.Context) *FileStream {
+	file, err := os.Open(file_name)
+	if err != nil {
+		log.Fatalf("Failed to open file '%s': %v", file_name, err)
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("Failed to start a watcher: %v", err)
+	}
+
+	err = watcher.Add(file_name)
+	if err != nil {
+		log.Fatalf("Failed to add file '%s' to the watcher: %v", file_name, err)
+	}
+
+	return &FileStream{
+		file_name: file_name,
+		reader:    file,
+		watcher:   watcher,
+		ctx:       ctx,
+		Done:      make(chan bool),
+	}
+}
+
+//-------------------------------------------------------------------------------------------------
+func (s *FileStream) Close() error {
+	err := s.watcher.Close()
+	if err == nil {
+		err = s.reader.Close()
+	}
+	return err
+}
+
+//-------------------------------------------------------------------------------------------------
+func (s *FileStream) WaitForChanges() (changed bool) {
+	result := make(chan bool)
+	go func() {
+		for {
+			select {
+			case event := <-s.watcher.Events:
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					result <- true
+					return
+				}
+			case <-s.Done:
+				result <- false
+				return
+			case <-s.ctx.Done():
+				result <- false
+				return
+			}
+		}
+	}()
+
+	return <-result
+}
+
+//-------------------------------------------------------------------------------------------------
+func (s *FileStream) readBlock(buffer []byte) int {
+	read_bytes, err := s.reader.Read(buffer)
+	if err == io.EOF || read_bytes == 0 {
+		return 0
+	}
+
+	if err != nil {
+		log.Fatalf("Failed to read data from file '%s': %v", s.file_name, err)
```

Comment:
`log.Fatal` will panic and exit the process, is that really what you want to do here?

Reply:
Doh, you're absolutely right and I forgot to update those. Fixed now.